plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/MongoQueryRunner.java

```diff
@@ -44,22 +47,36 @@ public static DistributedQueryRunner createMongoQueryRunner(MongoServer server,
 
     public static DistributedQueryRunner createMongoQueryRunner(MongoServer server, Map<String, String> extraProperties, Iterable<TpchTable<?>> tables)
             throws Exception
+    {
+        return createMongoQueryRunner(server, extraProperties, ImmutableMap.of(), ImmutableMap.of(), tables, runner -> {});
+    }
+
+    public static DistributedQueryRunner createMongoQueryRunner(
+            MongoServer server,
+            Map<String, String> extraProperties,
+            Map<String, String> coordinatorProperties,
+            Map<String, String> connectorProperties,
+            Iterable<TpchTable<?>> tables,
+            Consumer<QueryRunner> moreSetup)
+            throws Exception
     {
         DistributedQueryRunner queryRunner = null;
         try {
             queryRunner = DistributedQueryRunner.builder(createSession())
                     .setExtraProperties(extraProperties)
+                    .setCoordinatorProperties(coordinatorProperties)
+                    .setAdditionalSetup(moreSetup)
                     .build();
 
             queryRunner.installPlugin(new TpchPlugin());
             queryRunner.createCatalog("tpch", "tpch");
 
-            Map<String, String> properties = ImmutableMap.of(
-                    "mongodb.case-insensitive-name-matching", "true",
-                    "mongodb.connection-url", server.getConnectionString().toString());
+            connectorProperties = new HashMap(ImmutableMap.copyOf(connectorProperties));
```

Comment:
Why HashMap? I would stick to the pattern and use ImmutableMap.Builder here,

Reply:
`PostgreSqlQueryRunner` uses a HashMap, but OK.

Reply:
Using HashMap makes it possible to override the default connectorProperties. If i used an ImmutableMap.builder(), the builder doesn't have a `putIfAbsent`, and our style checks require i use `buildOrThrow()` rather than `build()`, so it'd be very awkward to allow that.